docs/other/osg-token-renewer.md

```diff
@@ -0,0 +1,174 @@
+----------------------------------------
+
+Install the OSG Token Renewal Service package:
+
+```console
+root@server # yum install osg-token-renewer
+```
+
+This will install the `osg-token-renewer` scripts & systemd service files,
+and will pull in the `oidc-agent` package that the service depends on.
+
+
+Configuring the OSG Token Renewal Service
+-----------------------------------------
+
+
+### Configuring accounts
+
+To create a new client account named `<ACCOUNT_SHORTNAME>`:
+
+1. Create a corresponding file named `/etc/osg/tokens/<ACCOUNT_SHORTNAME>.pw`
+   with the encryption password to use for this client account.
+1. Consult the
+   [Requesting Tokens](https://opensciencegrid.org/technology/software/requesting-tokens/)
+   document to determine which scopes you will need for this client account.
+1. Run the setup script as follows:
+
+        :::console
+        root@server # osg-token-renewer-setup.sh <ACCOUNT_SHORTNAME>
+
+    For example,
+
+        :::console
+        root@server # osg-token-renewer-setup.sh myaccount123
+
+1. When prompted, enter your Issuer and desired scopes for this account
+   from the list of valid options.
+1. You will also be prompted on the console to visit a web link to authorize
+   the client request with a passcode printed on the console.
+   Follow the prompts (visit the web link, enter the request passcode,
+   log in with your account for your issuer, and authorize the request).
+1. If this succeeds, you will be prompted with a new
+   `[account <ACCOUNT_SHORTNAME>]` section to add to your `config.ini`.
+   Add the section to your `/etc/token-renewer/config.ini`,
+   replacing the example section if it's still there.
+
+Next you can configure one or more tokens for this client account.
+
+
+### Configuring tokens
+
+After you have created an OIDC client account and added it to
+`/osg/token-renewer/config.ini`, you need to create a corresponding `token`
+section in the config for each token that should be generated for this account
+(possibly with different options).
+
+1.  Choose a `<TOKEN_NAME>` and add a new `[token <TOKEN_NAME>]` section
+    (replacing the example section if it's still there).
+
+    The `account` option in this section must match the `<ACCOUNT_SHORTNAME>`
+    for the corresponding `[account <ACCOUNT_SHORTNAME>]` section.
+
+1.  Set the `token_path` to
+    `/etc/osg/tokens/<ACCOUNT_SHORTNAME>.<TOKEN_NAME>.token`
+
+1.  Optionally, you may also specify any of the following options, which will
+    be passed to the `oidc-token` command when generating the token:
+
+    | Option         | Description                                               |
+    |:---------------|:----------------------------------------------------------|
+    | `audience`     | list of audiences to pass via `--aud` option              |
+    | `scope`        | list of scopes to pass via `--scope` option               |
+    | `min_lifetime` | min token lifetime in seconds to pass via `--time` option |
```

Comment:
```suggestion
1.  Optionally, you may also specify any of the following options, which can attenuate
    the respective values in the generated token compared to the associated with the account:

    | Option         | Description                                               |
    |:---------------|:----------------------------------------------------------|
    | `audience`     | list of audiences  (see [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3)) |
    | `scope`        | list of [scopes](https://github.com/WLCG-AuthZ-WG/common-jwt-profile/blob/master/profile.md#capability-based-authorization-scope)               |
    | `min_lifetime` | min token lifetime in seconds |
```

Reply:
> Optionally, you may also specify any of the following options, which can attenuate the respective values in the generated token compared to **the** associated **with the** account:

Should we change **the** to "those"?  Or drop the **with the** ?  It reads funny to me.

Also, "attenuate" is a big word to me that i had to look up.  Seems to mean "limit" ?

Reply:
> Should we change the to "those"? Or drop the with the ? It reads funny to me.

Yeah, that's better

> Also, "attenuate" is a big word to me that i had to look up. Seems to mean "limit" ?

Sure, sounds good